javascript/ql/experimental/adaptivethreatmodeling/modelbuilding/extraction/ExtractEndpointDataTraining.qll

```diff
@@ -73,6 +73,10 @@ query predicate trainingEndpoints(
       c instanceof LikelyNotASinkCharacteristic
     )
   ) and
+  // Don't surface endpoint filters as characteristics, because they were previously not surfaced.
+  // TODO: Experiment with surfacing these to the modeling code by removing the following line (and then make
+  // EndpointFilterCharacteristic private).
+  not characteristic instanceof EndpointFilterCharacteristic and
```

Comment:
OK, I see why your commit didn't fail `endpoint_large_scale/ExtractEndpointDataTraining.qlref`: It's because it didn't remove this line. If we were to raise the confidence of the endpoint filters without having this line, the training set would include all non-effective-sinks, which would be bad, because many of those are in fact low-confidence characteristics. 

Do you see a point in making `StandardEndpointFilterCharacteristic` private without removing this line or making `EndpointFilterCharacteristic` private?

IMO, I think we need to be consistent: Either we use subclasses of EndpointCharacteristics to replicate the existing data for now, or we try to improve the logic at the same time that we improve the code, and each PR requires end-to-end testing. I definitely prefer the former. The nice thing is that, once we're ready to progress to the second phase, even without my `TODO` comments and [issue tracking](https://github.com/github/ml-ql-adaptive-threat-modeling/issues/2126), we can easily look at all EndpointCharacteristics that aren't private and find the places they're used in the code, and that will tell use where we want to make improvements. Ultimately all EndpointCharacteristics should be private, with subclasses used only for two things: (1) Abstract classes are used to apply the same set of implications to many different subclasses without needing to overwrite `getImplications()` over and over. (2) Non-abstract classes are used to define the labels for type balancing through `getEndpoints()`.